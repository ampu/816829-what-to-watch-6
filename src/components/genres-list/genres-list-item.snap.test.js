import React from 'react';
import {Router} from 'react-router-dom';
import {createMemoryHistory} from 'history';

import {render} from '@testing-library/react';

import GenresListItem from './genres-list-item';

describe(`GenresListItem component`, () => {
  it(`Should render according to the snapshot while inactive`, () => {
    const {container} = render((
      <Router history={createMemoryHistory()}>
        <GenresListItem onGenreChange={jest.fn()} genre="fakeGenre"/>
      </Router>
    ));
    expect(container).toMatchSnapshot();
  });

  it(`Should render according to the snapshot while active`, () => {
    const {container} = render((
      <Router history={createMemoryHistory()}>
        <GenresListItem onGenreChange={jest.fn()} genre="fakeGenre" isActive/>
      </Router>
    ));
    expect(container).toMatchSnapshot();
  });
});
