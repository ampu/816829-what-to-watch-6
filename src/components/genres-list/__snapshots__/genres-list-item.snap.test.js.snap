// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenresListItem component Should render according to the snapshot while active 1`] = `
<div>
  <li
    class="catalog__genres-item catalog__genres-item--active"
  >
    <a
      class="catalog__genres-link"
      href="/"
    >
      fakeGenre
    </a>
  </li>
</div>
`;

exports[`GenresListItem component Should render according to the snapshot while inactive 1`] = `
<div>
  <li
    class="catalog__genres-item"
  >
    <a
      class="catalog__genres-link"
      href="/"
    >
      fakeGenre
    </a>
  </li>
</div>
`;
